lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/api/EvaluatorToSplitStrategy.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.io.data.loading.api;
+
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.reef.annotations.audience.DriverSide;
+import org.apache.reef.driver.catalog.NodeDescriptor;
+import org.apache.reef.io.data.loading.impl.DataPartition;
+import org.apache.reef.io.data.loading.impl.NumberedSplit;
+
+import java.util.Map;
+
+/**
+ * Interface that tracks the mapping between evaluators & the data splits
+ * assigned to those evaluators. Its part of the implementation of a
+ * {@link org.apache.reef.io.data.loading.api.DataLoadingService} that uses the
+ * Hadoop {@link org.apache.hadoop.mapred.InputFormat} to partition the data and
+ * request resources accordingly
+ *
+ * @param <V>
+ */
+@DriverSide
+public interface EvaluatorToSplitStrategy<V extends InputSplit> {
+
+  /**
+   * Initializes the mapping between partitions and splits.
+   * @param splitsPerPartition
+   *    the splits per partition
+   */
+  void init(Map<DataPartition, V[]> splitsPerPartition);
```

Comment:
That would be great to have but the InputFormatLoadingService first needs to compute the splits based on the partitions. Previous code did that, and then they hardcoded the creation of the "unique" strategy (it was just a class, no interface). 
Now we are injecting the strategy into InputFormatLoadingService. So the strategies constructors are empty, and then the init method is called. Kind of weird, but didn't want to keep on changing stuff.
Maybe there's another better way to do it without changing that much?